crankshaft-engine/src/task/resources.rs

```diff
@@ -134,23 +173,31 @@ impl From<&Defaults> for Resources {
     fn from(defaults: &Defaults) -> Self {
         Self {
             cpu: defaults.cpu(),
-            preemptible: Default::default(),
+            cpu_limit: defaults.cpu(),
             ram: defaults.ram(),
+            ram_limit: defaults.ram_limit(),
             disk: defaults.disk(),
+            preemptible: Default::default(),
             zones: Default::default(),
         }
     }
 }
 
 impl From<&Resources> for HostConfig {
     fn from(resources: &Resources) -> Self {
-        let mut host_config = HostConfig::default();
-        if let Some(ram) = resources.ram() {
-            host_config.memory = Some((ram * 1024. * 1024. * 1024.) as i64);
+        let mut host_config = Self::default();
+
+        // Note: Docker doesn't have a CPU reservation for containers
+        if let Some(cpu) = resources.cpu_limit() {
```

Comment:
I wonder if we should warn/debug here if the `Resources` contains these unfulfillable requirements?

Reply:
I think perhaps an `info` level; I feel like `warn` might be too noisy for what's basically something safe to ignore for local container execution.